File: server/php/src/Application/Domain/Payments/Mandate.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments;

use OnlinePayments\ExampleApp\Application\Domain\Enums\Payments\RecurrenceType;
use OnlinePayments\ExampleApp\Application\Domain\Enums\Payments\SignatureType;
use OnlinePayments\ExampleApp\Application\DTOs\Common\AddressDto;

class Mandate
{
    public ?string $iban = null;
    public ?string $customerReference = null;
    public ?string $mandateReference = null;

    public ?string $recurrenceType = null;
    public ?string $signatureType = null;

    public ?string $returnUrl = null;
    public ?AddressDto $address = null;

    public function __construct(
        ?string $iban = null,
        ?string $customerReference = null,
        ?string $mandateReference = null,
        ?string $recurrenceType = null,
        ?string $signatureType = null,
        ?string $returnUrl = null,
        ?AddressDto $address = null
    ) {
        $this->iban = $iban;
        $this->customerReference = $customerReference;
        $this->mandateReference = $mandateReference;
        $this->recurrenceType = $recurrenceType;
        $this->signatureType = $signatureType;
        $this->returnUrl = $returnUrl;
        $this->address = $address;
    }
}

File: server/python/app/application/domain/payments/mandate.py
from typing import Optional
from pydantic import BaseModel, ConfigDict
from pydantic.alias_generators import to_camel
from app.application.dtos.common.address_dto import AddressDto
from app.application.domain.enums.payments.recurrence_type import RecurrenceType
from app.application.domain.enums.payments.signature_type import SignatureType

class Mandate(BaseModel):
    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)

    iban: Optional[str] = None
    customer_reference: Optional[str] = None
    mandate_reference: Optional[str] = None
    recurrence_type: Optional[RecurrenceType] = None
    signature_type: Optional[SignatureType] = None
    return_url: Optional[str] = None
    address: Optional[AddressDto] = None